superduper/components/datatype.py

```diff
@@ -799,3 +857,40 @@ def get_serializer(
     'dill_lazy': dill_lazy,
     'file_lazy': file_lazy,
 }
+
+
+class Vector(BaseDataType):
+    """Vector meta-datatype for encoding vectors ready for search.
+
+    :param dtype: Datatype of encoded arrays.
+    """
+
+    identifier: str = ''
+    dtype: str = 'float64'
+
+    def __post_init__(self, db, artifacts):
+        self.identifier = f'vector[{self.shape[0]}]'
+        return super().__post_init__(db, artifacts)
+
+    @property
+    def encodable_cls(self):
+        return self.datatype_impl.encodable_cls
+
+    @cached_property
+    def datatype_impl(self):
+        if isinstance(CFG.datatype_presets.vector, str):
+            type_: str = CFG.datatype_presets.vector
+        else:
+            type_: str = self.db.databackend.datatype_presets['vector']
```

Comment:
I think using self.db.databackend in the datatype doesn’t seem ideal, but it can be acceptable as a temporary solution if it’s just a hotfix.

Reply:
I think it's important to have defaults which don't need to be configured.

Reply:
I haven’t come up with a better way to configure the default backend type for now, so let’s use this method